### PR TITLE
Gameplay 3: Turret

### PR DESCRIPTION
**Pull request by BohanXu**
-Sentinel (MultishotTurret) now can consecutively shoot. Need Repaint! 

Sentinel now doesn't shoot simultaneously. 

- Sentinel now can be toggle as stationary or not.

-We might need look ahead aiming?

### CODE REVIEW

File: Assets/Scripts/Enemy/Turret.cs [hunk 1]
```diff
@@ -63,17 +79,31 @@ private void PointAtPlayer()
...
+        //We add a small noise around 
+        actualCD = cooldownTime + (Random.value - 0.5f) * 2 * cdNoisePercent;
 
-        // Play shoot sound
-        audioSource.Play();
     }
 
+    IEnumerator ConsecutiveShoot()
+    {
+        // Create bullet instance, set rotation
```
Comment (makersmelx) [inline Assets/Scripts/Enemy/Turret.cs:96]: won't you set `isTurnable` false here?

File: Assets/Scripts/Enemy/Turret.cs [hunk 2]
```diff
@@ -63,17 +79,31 @@ private void PointAtPlayer()
...
+        for (int i = 0; i < numBullet; i++)
+        {
+            Bullet bulletRef = Instantiate(bulletPrefab, gunHeadTransform.position, transform.rotation);
+            bulletRef.SetBulletRotation(gunTransform.rotation);
+
+            // Play shoot sound
+            audioSource.Play();
+            yield return new WaitForSeconds(timeInBetweenBullet);
```
Comment (makersmelx) [inline Assets/Scripts/Enemy/Turret.cs:104]: the actual cooldown here is `timeBetweenBuleet+cooldownTime`

Comment (makersmelx) [reply Assets/Scripts/Enemy/Turret.cs:104]: i think it will be tricky after one week. just make it clear here

**Review by makersmelx — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Enemy/Turret.cs (hunk 2 above)
Comment (BohanXu) [reply Assets/Scripts/Enemy/Turret.cs:104]: Nah it's irrelevant to cooldownTime. timeInBetweenBullet defines the time between each shot of a multishot

File: Assets/Scripts/Enemy/Turret.cs (hunk 1 above)
Comment (BohanXu) [reply Assets/Scripts/Enemy/Turret.cs:96]: The turret would still follows player

**Pull request merged**